[Xcode] Dead webkitdirs.pm code assumes all signing identities start with "iPhone Developer"

emw-apple opened the pull request:
#### 3ad1996f37dc12b3b72f5e587a7d46d252bb4430
<pre>
[Xcode] Dead webkitdirs.pm code assumes all signing identities start with &quot;iPhone Developer&quot;
<a href="https://bugs.webkit.org/show_bug.cgi?id=263938">https://bugs.webkit.org/show_bug.cgi?id=263938</a>
<a href="https://rdar.apple.com/96538743">rdar://96538743</a>

Reviewed by Alexey Proskuryakov.

They have been called &quot;Apple Developer&quot; for years now, so the code path
that disables code signing is always taken. Since it&apos;s not possible to
run open-source builds of WebKit on embedded devices, this has never
been noticed, and it can be cleaned up.

Move the code signing overrides into xcconfig variables that are only
active in open source builds. This has the nice side effect of making it
possible to build for embedded devices in Xcode, without using
build-webkit.

Works towards making webkitdirs aware of the active SDK in internal
builds (<a href="https://rdar.apple.com/86340301).">rdar://86340301).</a>

* Source/JavaScriptCore/Configurations/DebugRelease.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/DebugRelease.xcconfig:
* Source/WebKit/Configurations/DebugRelease.xcconfig:
* Tools/DumpRenderTree/mac/Configurations/DebugRelease.xcconfig:
* Tools/MobileMiniBrowser/Configurations/DebugRelease.xcconfig:
* Tools/Scripts/webkitdirs.pm:
(XcodeOptions):
(hasIOSDevelopmentCertificate): Deleted.
* Tools/TestWebKitAPI/Configurations/DebugRelease.xcconfig:
* Tools/WebKitTestRunner/Configurations/DebugRelease.xcconfig:

Canonical link: <a href="https://commits.webkit.org/270072@main">https://commits.webkit.org/270072@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cf724a125c86172c6bb9b850f43e1fde71d8174b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24391 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/2587 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25505 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26522 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22439 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/24662 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/4222 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/59 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22865 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24637 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/2008 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/21068 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/27108 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1746 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21994 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/28205 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/21220 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22232 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22309 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25996 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/23684 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1687 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/33 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/31082 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/2973 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/6814 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5860 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/2101 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/31048 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/2056 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/6490 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/JavaScriptCore/Configurations/DebugRelease.xcconfig:
```diff
@@ -34,6 +34,9 @@ CODE_SIGN_IDENTITY_ = $(CODE_SIGN_IDENTITY_NO);
 CODE_SIGN_IDENTITY_NO = -;
 CODE_SIGN_IDENTITY_YES = $(WK_ENGINEERING_CODE_SIGN_IDENTITY);
 
+CODE_SIGN_IDENTITY[sdk=embedded] = $(CODE_SIGN_IDENTITY_EMBEDDED_$(USE_INTERNAL_SDK));
+CODE_SIGN_IDENTITY_EMBEDDED_YES = $(inherited);
```
This seems inconsistent with `macos` case above. We set it to `-` there, and make it empty here.

emw-apple replied:
Right—if we tell the public iOS SDK to ad-hoc sign, it fails with:
```
error: Ad Hoc code signing is not allowed with SDK 'iOS 17.0'
```

So this configuration disables code signing altogether. (And it matches what webkitdirs used to do: `push @options, "CODE_SIGN_IDENTITY="; # No identity`)

emw-apple committed:
[Xcode] Dead webkitdirs.pm code assumes all signing identities start with "iPhone Developer"
https://bugs.webkit.org/show_bug.cgi?id=263938
rdar://96538743

Reviewed by Alexey Proskuryakov.

They have been called "Apple Developer" for years now, so the code path
that disables code signing is always taken. Since it's not possible to
run open-source builds of WebKit on embedded devices, this has never
been noticed, and it can be cleaned up.

Move the code signing overrides into xcconfig variables that are only
active in open source builds. This has the nice side effect of making it
possible to build for embedded devices in Xcode, without using
build-webkit.

Works towards making webkitdirs aware of the active SDK in internal
builds (rdar://86340301).

* Source/JavaScriptCore/Configurations/DebugRelease.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/DebugRelease.xcconfig:
* Source/WebKit/Configurations/DebugRelease.xcconfig:
* Tools/DumpRenderTree/mac/Configurations/DebugRelease.xcconfig:
* Tools/MobileMiniBrowser/Configurations/DebugRelease.xcconfig:
* Tools/Scripts/webkitdirs.pm:
(XcodeOptions):
(hasIOSDevelopmentCertificate): Deleted.
* Tools/TestWebKitAPI/Configurations/DebugRelease.xcconfig:
* Tools/WebKitTestRunner/Configurations/DebugRelease.xcconfig:

Canonical link: https://commits.webkit.org/270072@main
